### ISSUE TITLE
[Enhancement] Master asset packs

### ISSUE THREAD

**Issue by P-Star7** [TheXTech/TheXTech#348]
In the current TheXTech and SMBX 1.3, there is the "graphics", "music", and "sounds" folders. The issue is that some episodes rely on replacements for those graphics, sounds, and musics that are not in the episode's folder. Here is a partial list:
-Super Talking Time Bros. 1, Super Talking Time Bros 2., Super Talking Time Bros. 2.5: The Last Levels
-A Super Mario Bros. X Thing: Prelude to the Stupid
-A Super Mario Bros. X Thing 2: Analog Funk
-Love Bodhi's upcoming Nostalgic Paradise episodes
All of these four sets of episodes require different graphics, sounds, and musics than the default SMBX 1.3 levels that are very common. So here is my proposal.

Here is a mockup of how the TheXTech folder would be like
Root Directory
-assets
--SMBX64
---graphics
---music
---sounds
--STTB
---graphics
---music
---sounds
--ASMBXT
---graphics
---music
---sounds
--ASMBXT2E1
---graphics
---music
---sounds

This also requires support for reading the npc.txt codes from these folders, for the ASMBXT/ASMBXT2E1 NPCs (framestyle=1 goomba), player's .inis for Nostalgic Paradise hitboxes, and music.ini/sounds.ini would be nice to have as well.

The way that TheXTech would recognize assets is by information in the .wldx or .lvlx file with two fields.
1. String (name of the asset pack that is required)
2. Integer (minimum version of the asset pack that is required)
If any of these do not have a qualifying asset pack in the assets folder, then the episode will abort with an error message and return to the TheXTech title screen.

Once TheXTech reads the asset information from a .lvl(x) or .wld(x), it checks into that subfolder of the assets folder. So for example, if the player is playing Nostalgic Paradise, it sees that the .wldx says to refer to assets/NostalgicParadise for all graphics, musics, and sounds.

This would require the .wld files for STTB, STTB2, STTB2.5, ASMBXT, and ASMBXT2E1 to be updated to .wldx manually and have this info added to them, but it's a small thing to do to be able to play all of these episodes in the same installation of TheXTech.

And of course, any .lvl or .wld that does not have this info in it, which is most old levels and episodes, defaults to the SMBX64 asset pack without requiring manual conversion.

As a bonus, putting this information into .lvlx files allows for standalone levels to be integrated with asset packs, once standalone level running is added to the title screen.

As yet another bonus, this allows for a new "non-infringement" asset pack of graphics, music, and sounds to be distributed with TheXTech without compromising the player's ability to also play Mario-themed episodes in the same installation of TheXTech. All they must do is put the "SMBX64" asset pack in the assets folder.

In short, TheXTech would look for resources in this order:
1. Folder of the current level
2. Folder above the current level
3. Otherwise, look for assets in TheXTech's root folder/assets/[STRING WHICH IS IN THIS .WLDX OR .LVLX FILE]

This third step was formerly "look in TheXTech's root folder/graphics, music, and sounds."

**Comment by 0lhi** [comment]
Duplicate of #291. Gonna close this.

If you have any additional or alternative suggestions for asset pack handling, please comment there.
